src/main/scala/viper/gobra/reporting/StatsCollector.scala

```diff
@@ -375,6 +375,19 @@ case class StatsCollector(reporter: GobraReporter) extends GobraReporter {
           isBuiltIn = isBuiltIn)
       // Consider the enclosing function, for closure declarations
       case p: PClosureDecl => getMemberInformation(nodeTypeInfo.enclosingFunctionOrMethod(p).get, typeInfo, viperMember)
+      
+      case p: PDerefDecl =>
+        GobraMemberInfo(
+          pkgId = pkgId,
+          pkg = pkgName,
+          memberName = s"${p.typ}" ++ "_deref",
+          args = "",
+          nodeType = FunctionDeclaration,
+          hasSpecification = true,
+          isTrusted = false,
+          isAbstractAndNotImported = false,
+          isImported = isImported,
+          isBuiltIn = isBuiltIn)
```

Comment:
PDerefDecl needs to be extended with `PCodeRootWithResult`, since it is a pure function that returns a value. Although the PConstructDecl and PAssignDecl also has a return value, it will only be generated in the encoding. If `PCodeRootWithResult` gets removed from PDerefDecl, it would fail with: `error: Statement does not root in a CodeRoot`. 

However, after implementation it would throw the error:
```
...
[error]         at viper.gobra.reporting.StatsCollector.getMemberInformation(StatsCollector.scala:392)
[error]         at viper.gobra.reporting.StatsCollector.getMemberInformation(StatsCollector.scala:392)
[error]         at viper.gobra.reporting.StatsCollector.getMemberInformation(StatsCollector.scala:392)
...
```
It would continuously throw this error until the terminal is stopped. Therefore, the StatsCollector needs to be defined.

Reply:
I understand now why deref is necessary, but it still illustrates to me that cases for the assignment and constructor should be added as well (even if it does not crash when they are missing).